crates/bevy_ptr/src/lib.rs

```diff
@@ -397,13 +397,20 @@ impl<'a, T: ?Sized> From<&'a mut T> for PtrMut<'a> {
 }
 
 impl<'a> OwningPtr<'a> {
+    /// This exists mostly to reduce compile times;
+    /// code is only duplicated per type, rather than per function called.
+    fn make_internal<T>(temp: &mut ManuallyDrop<T>) -> OwningPtr<'_> {
```

Comment:
This should be an unsafe method and have the same safety requirements as the promote method.